### ISSUE TITLE
Add .col-xs-0 .col-sm-0 .col-md-0 and .col-lg-0 (or at least generic col-0) for animation purposes

### ISSUE THREAD

**Issue by McNetic** [twbs/bootstrap#18416]
As #11391 is locked, I can not comment on it or reopen it, so I have to add a new issue:

I second the addition of the above classes (or at least a generic col-0), but for completely different reasons than already advocated: I like to change the width of my elements with javascript, for example, I have two columns with width 4 and 8 respectively, and want to hide the first one while making the second one full width. This does not work nicely with hidden-*.

Have a look at this http://jsfiddle.net/McNetic/zd4bzd9k/4/ . It shows the behaviour with hidden-* and also with the proposed col-0.

**Comment by twbs-lmvtfy** [comment]
Hi @McNetic!

You appear to have posted a live example (http://fiddle.jshell.net/McNetic/zd4bzd9k/3/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 102, column 5: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 94, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 102, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by McNetic** [comment]
While it is not relevant, I fixed the issues http://jsfiddle.net/McNetic/zd4bzd9k/3/.

**Comment by twbs-lmvtfy** [comment]
Hi @McNetic!

You appear to have posted a live example (http://fiddle.jshell.net/McNetic/zd4bzd9k/3/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 102, column 5: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 94, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 102, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by McNetic** [comment]
Hmpf. http://jsfiddle.net/McNetic/zd4bzd9k/4/

**Comment by cvrebert** [comment]
Port of the example to Bootstrap 4: http://jsfiddle.net/cvrebert/m1k7hxbj/

**Comment by McNetic** [comment]
Which still exposes the undesired behaviour (I first thought the port would show how this is fixed in Bootstrap 4).
A small side note: This simple example would also work with css transition (given something like col-0 is implemented), but as soon as one of the divs contains content (like in my use case, Highchart charts) that is dependent on resize events to adapt to the div size, it breaks (aside from the content size change not being animated anyways, which I have to do with the jquery animation progress event).

**Comment by cvrebert** [comment]
> as soon as one of the divs contains content (like in my use case, Highchart charts) that is dependent on resize events to adapt to the div size, it breaks

Yup, that's par for the course. Need to issue a redraw/reinitialize command to the library when you mess with its container's dimensions. AFAIK, browsers don't offer any sort of "element dimensions changed" event that Highcharts could hook into. The Web Platform is sad.

**Comment by cvrebert** [comment]
> Is there a chance this will also be added to Bootstrap 3 (in case it is accepted at all)?

Negatory. Please refer to https://github.com/twbs/bootstrap/blob/master/CONTRIBUTING.md#feature-requests

**Comment by mdo** [comment]
I don't think we'll be implementing this at all—seems solvable without additional grid classes on our end. Thanks though!

**Comment by McNetic** [comment]
Huh? How can this be solved without additional classes?

**Comment by cvrebert** [comment]
> seems solvable without additional [...] classes on our end. 

@mdo I mean, you can say that about almost anything. Do you mean that you don't believe it's a common/important enough problem to include a solution in the framework itself?

**Comment by mdo** [comment]
> Huh? How can this be solved without additional classes?

I mean you don't explicitly need more grid classes to achieve the same effect. You can do it by nuking the width with JavaScript as opposed to toggling a class, for example.

> I mean, you can say that about almost anything. Do you mean that you don't believe it's a common/important enough problem to include a solution in the framework itself?

Doesn't seem generalized enough for the framework, in particular our grid system.

**Comment by McNetic** [comment]
If you want to have the effect without javascript, just with css transitions (which works for simpler cases and is a valid use case from my point of view), you'll need a class. If you don't want to throw around css styles in javascript files, you'll need a class. If you want to change some other parameters also (like the height in my case, I want the charts to go big for better inspection), it get's messy without using classes. If you want different behaviour based on screen size (responsiveness), you'll need a class.

So I think there are some reasons to implement a class, but I don't want to push you, just provide some more aspects to the topic.

**Comment by mdo** [comment]
I'm not saying you don't need a class, or that one would be useful. I'm saying I don't want to add even more classes to our grid system to suit one example. All that means is, I don't see a strong enough reason to add these classes to Bootstrap. I'd encourage you to extend the grid system yourself with those classes if you're feeling up to it.
